cmd/kafka_prod_receiver/kafka_prod_receiver.go

```diff
@@ -2,44 +2,122 @@ package main
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
+	"log"
+	"time"
 
 	"github.com/cybertec-postgresql/pgwatch/v3/api"
 	"github.com/destrex271/pgwatch3_rpc_server/sinks"
 	"github.com/segmentio/kafka-go"
 )
 
 type KafkaProdReceiver struct {
-	conn      *kafka.Conn
-	topic     string
-	partition int
-	uri       string
+	conn_regisrty map[string]*kafka.Conn
+	uri           string
+	auto_add      bool
 	sinks.SyncMetricHandler
 }
 
-func NewKafkaProducer(host string, port string, topic string, partition int) (kpr KafkaProdReceiver, err error) {
-	var conn *kafka.Conn
-	conn, err = kafka.DialLeader(context.Background(), "tcp", host+":"+port, topic, partition)
-	if err != nil {
-		return
+func (r *KafkaProdReceiver) HandleSyncMetric() {
+	req := <-r.SyncChannel
+	if req.Operation == "DELETE" {
+		r.CloseConnectionForDB(req.DbName)
+	} else if req.Operation == "ADD" {
+		r.AddTopicIfNotExists(req.DbName)
```

Comment:
```suggestion
	switch req.Operation {
	   "DELETE": r.CloseConnectionForDB(req.DbName)
	   "ADD": r.AddTopicIfNotExists(req.DbName)
	}
```

How about this shorter notation? It's not an error but the style :)

Reply:
Looks good!